### PR TITLE
Revert smart pointer type change in `PointCloud` and deprecate `getMapping()`

### PR DESCRIPTION
**Pull request by taketwo**
Even though `pcl::detail::getMapping()` is not officially a public API, it is used outside of PCL (in `pcl_ros`). Changing the smart pointer type from boost to std broke that code. (More information in #2792.)

This commit partially reverses the changes from #3474, unbreaking `pcl_ros`. It also properly deprecates the function so that other (unknown to us) clients have time to update their code.

### CODE REVIEW

File: common/include/pcl/point_cloud.h [hunk 1]
```diff
@@ -603,17 +606,17 @@ namespace pcl
 
     protected:
       // This is motivated by ROS integration. Users should not need to access mapping_.
-      std::shared_ptr<MsgFieldMap> mapping_;
+      [[deprecated("rewrite your code to avoid using it")]] boost::shared_ptr<MsgFieldMap> mapping_;
```
Comment (kunaltyagi) [inline common/include/pcl/point_cloud.h]: Maybe reword the message to include what to do instead. Like `Use pcl::createMapping(vector<pcl::PCLPointField>, pcl::MsgFieldMap&)` instead

Comment (taketwo) [reply common/include/pcl/point_cloud.h]: I also had this thought but was not sure since it's not a true replacement. But since you also think this way, I'll update the message.

Comment (kunaltyagi) [reply common/include/pcl/point_cloud.h]: It's not public, but since there was a public usage, people (maintainers) who compile from source (and check warnings 😅 ) will know

Comment (kunaltyagi) [reply common/include/pcl/point_cloud.h]: I like the new sassy one 🤣

Comment (taketwo) [reply common/include/pcl/point_cloud.h]: Well yeah, sounds a bit presumptuous, but what else can I say :D Here it's not about how to create a mapping, but rather about not using PCL's protected space to store your stuff.
